alpha/lib/model/FileSync.php

```diff
@@ -271,7 +271,27 @@ public function setEncryptionKey ($v) { $this->putInCustomData("encryptionKey",
 	public function isEncrypted () { return ($this->getFromCustomData("encryptionKey"))? true : false ; }
 	public function getIv() {return kConf::get("encryption_iv");}
 
+	/**
+	 * Create new fileSync With status pending and new storageId
+	 * @param $storageId
+	 * @return FileSync
+	 * @throws PropelException
+	 */
+	public function createFileSyncCopyForStorageExport($storageId)
+	{
+		$newfileSync = $this->copy(true);
+		$newfileSync->setStatus(FileSync::FILE_SYNC_STATUS_PENDING);
+		$newfileSync->setSrcPath($this->getFullPath());
+		$newfileSync->setSrcEncKey($this->getSrcEncKey());
+		$newfileSync->setFileType(FileSync::FILE_SYNC_FILE_TYPE_URL);
+		$newfileSync->setDc($storageId);
 
+		$fileSyncKey = kFileSyncUtils::getKeyForFileSync($newfileSync);
+		list($root, $filePath) = kPathManager::getFilePathArr($fileSyncKey, $storageId);
+		$newfileSync->setFilePath($filePath);
+		$newfileSync->save();
```

Comment:
The save better be done externally as this function is only creating an object. The caller needs to decide if to save it to the DB or maybe change something before doing it.

Reply:
k